components/automate-ui/src/app/pages/+compliance/+reporting/+reporting-overview/overview-trend/overview-trend.component.ts

```diff
@@ -51,9 +52,13 @@ export class OverviewTrendComponent implements OnChanges, OnDestroy  {
   }
 
   get trendData(): TrendData[] {
-    return this.data.map(d => {
-      return { ...d, report_time: this.createUtcDate(d.report_time) };
-    });
+    if(this.treandDataCache === undefined || this.treandDataCache.length == 0) {
+      console.log("trendDataCache..!")
+      this.treandDataCache = this.data.map(d => {
+        return { ...d, report_time: this.createUtcDate(d.report_time) };
+      });
+    }
+    return this.treandDataCache
   }
 
   get domainX() {
```

Comment:
Outside the scope of your task here, I would urge you to examine every getter in this file to see which could be changed to private like this. I checked about 10 and every single one should be private.
```suggestion
  private get domainX() {
```